### PR TITLE
#137 handle invalid jwt

### LINKED ISSUE #137: Handle invalid JWT
Invalid JWT is not handled cleanly. It would be a good idea to add a middleware to handle generic errors (and add an error for incorrectly formatted or invalid JWT).

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #137. Branch off #115.

### CODE REVIEW

File: src/server/routes/masterlist.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+/* eslint-disable no-console */
+
+export const getMasterlist = ({ db, secret }) => async (req, res) => {
+  const token = req.getJWT();
+  if (token && token.isValid(secret)) {
+    const storeId = token.getField('storeId');
+
+    try {
+      const masterlists = await db.getMasterLists(storeId);
```
Comment (joshxg) [inline src/server/routes/masterlist.js]: I think this should be db.getMasterlists

**Comment by joshxg** [comment]
Looks good, will wait for #115 and conflicts

File: src/server/routes/masterlist.js (hunk 1 above)
Comment (wlthomson) [reply src/server/routes/masterlist.js]: Good spot. Fixed.

**Comment by wlthomson** [comment]
Typo and conflicts resolved. @joshxg As you stated earlier, probably best to merge this after #115.

**Comment by wlthomson** [comment]
@joshxg All ready :).

**Pull request merged**